openedx/features/colaraz_features/auth_backend.py

```diff
@@ -69,3 +71,22 @@ def get_user_details(self, response):
     def _id3_config(self):
         from third_party_auth.models import OAuth2ProviderConfig
         return OAuth2ProviderConfig.current(self.name)
+
+    def user_data(self, access_token, *args, **kwargs):
+        """
+        Returns the user data receive by social backend and register in case of user doesn't exists already.
+        """
+
+        # Imported here to avoid circular imports
+        from third_party_auth.utils import user_exists
+
+        user_data = super(ColarazIdentityServer, self).user_data(access_token, *args, **kwargs)
+        request = get_current_request()
+        request_source = request.GET.get('src')
+
+        if (request.view_name == "AccessTokenExchangeView" and request_source == 'mobile'
```

Comment:
its better according to [pep8](https://stackoverflow.com/a/53173/5579792) to end with a logical operator

Reply:
Done